src/components/App/index.tsx

```diff
@@ -74,18 +80,28 @@ export default class App extends Component<Props, State> {
   }
 
   @bind
-  private onBack() {
-    this.setState({ file: undefined });
+  private onPopState() {
+    this.setState({ isEditorOpen: history.pathname === ROUTE_EDITOR });
+  }
+
+  @bind
+  private openEditor() {
+    history.push(ROUTE_EDITOR);
```

Comment:
Maybe there's a better way, but I guess this could be done in one of the state-change callbacks, as in if `state.file` goes from undefined to defined, then push state. That would avoid pushing state when the file is just changed.

Reply:
I added a check, if `isEditorOpen` is true then we don't do anything, since the file will change by itself, and we don't need to update the history